Sources/SWBUtil/Lock.swift

```diff
@@ -26,6 +26,9 @@ public final class Lock: @unchecked Sendable {
     #if os(Windows)
     @usableFromInline
     let mutex: UnsafeMutablePointer<SRWLOCK> = UnsafeMutablePointer.allocate(capacity: 1)
+    #elseif os(OpenBSD)
+    @usableFromInline
+    let mutex: UnsafeMutablePointer<pthread_mutex_t?> = UnsafeMutablePointer.allocate(capacity: 1)
```

Comment:
Do you know why there's this difference between BSDs and Darwin/Linux? I needed it for FreeBSD as well.

Can we safely just use the Optional variant on all platforms?

Reply:
This happens because OpenBSD's value of `pthread_t` and subsidiary types like `pthread_mutex_t` is a pointer, whereas on Linux for example, it's typedef'd to an int (and `pthread_mutex_t` to a union). Of course, there's no nullability annotations on the 'BSDs, so Swift is interpreting the type as an optional like it would any other nullable pointer. 

After thinking about it briefly I think you have an incongruity either way, so I don't think you can use Optional always: you either have to declare, for example, `mutex` as Optional here and then conditionally check the optional or not depending on platform at the callsite to `pthread_mutex_init`, or you do what we do here, and conditionally declare `mutex` as Optional or not.